### PR TITLE
Add otp sign endpoint

### PR DESCRIPTION
**Pull request by maksim-litvinov**
Relates to https://github.com/rubykube/barong/issues/151

### CODE REVIEW

File: app/api/management_api/v1/otp.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        end
+        params do
+          requires :account_uid, type: String, allow_blank: false, desc: 'Account UID'
+          requires :otp_code, type: Integer, allow_blank: false, desc: 'Code from Google Authenticator'
+          requires :request_data, type: Hash, allow_blank: false, desc: 'Request data from App Logic'
+        end
+        post '/sign' do
+          declared_params = declared(params)
+          account = Account.kept.active.find_by!(uid: declared_params[:account_uid])
+          error!('Account has not enabled 2FA', 422) unless account.otp_enabled
```
Comment (yivo) [inline app/api/management_api/v1/otp.rb]: Use `otp_enabled?`

File: app/api/management_api/v1/otp.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          require 'openssl'
+          keychain = { barong: Barong::Security.private_key }
+          algorithms = { barong: 'RS256' }
+          JWT::Multisig.generate_jwt(payload, keychain, algorithms)
+        end
+      end
+
+      desc 'OTP related routes'
+      resource :otp do
+        desc "Sign request with barong signature" do
```
Comment (yivo) [inline app/api/management_api/v1/otp.rb]: Don't mix quotes.

File: app/api/management_api/v1/otp.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'barong/security/access_token'
+
+module ManagementAPI
+  module V1
+    class OTP < Grape::API
+      helpers do
+        def generate_signed_payload(payload)
+          require 'openssl'
```
Comment (yivo) [inline app/api/management_api/v1/otp.rb]: Require dependencies in eager load stage.

File: app/api/management_api/v1/otp.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          keychain = { barong: Barong::Security.private_key }
+          algorithms = { barong: 'RS256' }
+          JWT::Multisig.generate_jwt(payload, keychain, algorithms)
```
Comment (yivo) [inline app/api/management_api/v1/otp.rb]: Try to modify existing JWT using https://github.com/rubykube/jwt-multisig/blob/master/lib/jwt-multisig.rb#L65

File: app/api/management_api/v1/otp.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+      desc 'OTP related routes'
+      resource :otp do
+        desc "Sign request with barong signature" do
+          @settings[:scope] = :otpsign
+        end
+        params do
+          requires :account_uid, type: String, allow_blank: false, desc: 'Account UID'
+          requires :otp_code, type: Integer, allow_blank: false, desc: 'Code from Google Authenticator'
+          requires :request_data, type: Hash, allow_blank: false, desc: 'Request data from App Logic'
```
Comment (yivo) [inline app/api/management_api/v1/otp.rb]: You can accept complete JWT here and add signature.

File: app/api/management_api/v1/otp.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+      end
+
+      desc 'OTP related routes'
+      resource :otp do
+        desc "Sign request with barong signature" do
+          @settings[:scope] = :otpsign
+        end
+        params do
+          requires :account_uid, type: String, allow_blank: false, desc: 'Account UID'
+          requires :otp_code, type: Integer, allow_blank: false, desc: 'Code from Google Authenticator'
```
Comment (yivo) [inline app/api/management_api/v1/otp.rb]: Maybe it is better to keep this as string not integer?

**Comment by maksim-litvinov** [comment]
Tested with Applogic

**Pull request merged**